### PR TITLE
[14.0][MIG] base_delivery_carrier_files

### PR DESCRIPTION
**Pull request by StefanRijnhart**

### CODE REVIEW

File: base_delivery_carrier_files/models/stock_move.py
Comment (hailangvn) [inline base_delivery_carrier_files/models/stock_move.py:16]: May I know the benefit of using filtered+lambda over search+domain? Thanks

**Review by hailangvn — APPROVED** [review]
Code review: LGTM. Thank you.

File: base_delivery_carrier_files/models/stock_move.py
Comment (hailangvn) [reply base_delivery_carrier_files/models/stock_move.py:16]: Example: https://github.com/odoo/odoo/blob/c08082d89782b7675e8e649cba15ca9e03e40077/addons/purchase_stock/tests/test_replenish_wizard.py#L51

Comment (StefanRijnhart) [reply base_delivery_carrier_files/models/stock_move.py:16]: Good question! Because the processes of pickings and their stock moves are very much intertwined, it is likely that at this point we will already have the picking states in our cache so then we avoid the search as well as a potential read. Also, when setting the state of more than one stock move, it is unlikely that there are a large number of pickings involved. I would not prefer to use `filtered` if the result set was potentially large (but it also depends on whether the column used in the domain is indexed, which happens to be the case here).

**Comment by silverf0x0** [comment]
Hi there. What is missing here. Any help needed?

**Comment by StefanRijnhart** [comment]
@silverf0x0 this is a standard migration PR (and a tiny API change when storing the output, in the last commit). It's waiting for a second review.

**Comment by silverf0x0** [comment]
@StefanRijnhart Understood. Thank you.

**Comment by StefanRijnhart** [comment]
@silverf0x0 by the way, that review could come from you if you have the time!

**Comment by silverf0x0** [comment]
@StefanRijnhart Nice to ear that, because in this very moment I'm testing your work. Any specifc procedure is needed for the review, or I just need to be sure that everything is working?

**Comment by StefanRijnhart** [comment]
@silverf0x0 Great, thanks! If you successfully tested the module, please mention that in your approval comment. No specific procedure applies.

**Comment by silverf0x0** [comment]
@StefanRijnhart Excellent, so I'll finish all the tests and I'll approve the review. Thanks to you!

**Comment by silverf0x0** [comment]
@StefanRijnhart I've completed to test everything and for me work/look nice. Maybe the only thing we could add, as improvement, is a UserError (or ValidationError) withing models/stock_picking.py, within generate_carrier_files function:

- when a carrier is not present in an outgoing picking it would not generate a file, as per standard. In this case would be useful to let the user know why a file is not being generated (but in this case the raise could block the generation of other files, which is a little bit annoying). Maybe we can just leave a to-do in the comment?

Thank you in advance

**Review by silverf0x0 — APPROVED** [review]
Tested locally several times, without issues.
Code has been reviewed and is fine.

**Review by silverf0x0 — APPROVED** [review]

File: base_delivery_carrier_files/__manifest__.py
Comment (gfcapalbo) [inline base_delivery_carrier_files/__manifest__.py:11]: in v14 delivery implies stock
delivery -> sale_stock ->  stock_account -> stock

File: base_delivery_carrier_files/models/stock_picking.py
Comment (gfcapalbo) [inline base_delivery_carrier_files/models/stock_picking.py:12]: fails silently when a picking with  no carrier or no carrier.carrier_file_id  is selected.
Perhaps some feedback (like a warning icon), or  just filtering out non-eligible stock.picking from wizard could be a usability improvement.

**Review by gfcapalbo — APPROVED** [review]
Tested locally, generated the CSV, all looks good , just some observations on usability, those can be discarded because the module will be used by other modules that can add usability niceties themselves.
LGTM.

**Pull request merged**
